### PR TITLE
TPROD-321 - NotificationBundle - Mobile - Convert PHP Templates to Twig Templates

### PR DESCRIPTION
**Pull request by JoshuaEstes**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Sane testing steps of #11908 

- `/s/mobile_notifications` List
  - Does page load and render correctly with and without any notifications present
  - Sorting columns work correctly
  - Filtering work correctly
- `/s/mobile_notifications/new` New
  - Can new notifications be created
- `/s/mobile_notifications/view/{id}` Details
  - Page load/render correctly with no errors
- `/s/mobile_notifications/edit/{id}` Details
  - Can you edit the notification

### CODE REVIEW

File: app/bundles/NotificationBundle/Views/MobileNotification/list_graph.html.twig [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{#
```
Comment (JoshuaEstes) [inline app/bundles/NotificationBundle/Views/MobileNotification/list_graph.html.twig:1]: This file appears to not be used anywhere in the code

File: app/bundles/NotificationBundle/Views/MobileNotification/template_graph.html.twig [hunk 2]
```diff
@@ -1,18 +1,7 @@
-<?php
```
Comment (JoshuaEstes) [inline app/bundles/NotificationBundle/Views/MobileNotification/template_graph.html.twig:1]: This file appears to not be used anywhere in the code

**Comment by RCheesley** [comment]
@JoshuaEstes it would be helpful to have a steer as to what you need us to test with this PR, it's not obvious from the PR.

**Review by mabumusa1 — COMMENTED** [review]
- [X] List of Notifications 
- [X] Create a New Notification
- [X] Create New Category 
- [X] Notification View
- [X] Notification Edit 
- [X] Search 

The same error shows in [here](https://github.com/mautic/mautic/pull/11908#pullrequestreview-1282472715)  happens here.

Another issue, try to create a new notification and keep the `message` field empty that validation will not happen which causes the page to fail with this log

```[2023-02-03 08:15:14] mautic.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\Exception\NotNullConstraintViolationException: "An exception occurred while executing 'INSERT INTO push_notifications (is_published, date_added, created_by, created_by_user, date_modified, modified_by, modified_by_user, checked_out, checked_out_by, checked_out_by_user, name, description, lang, url, heading, message, button, utm_tags, notification_type, publish_up, publish_down, read_count, sent_count, mobile, mobileSettings, category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1, "2023-02-03 08:15:14", 1, "Admin Mautic", null, null, null, null, null, null, "test", null, "en", null, "test", null, null, "a:4:{s:9:\"utmSource\";N;s:9:\"utmMedium\";N;s:11:\"utmCampaign\";N;s:10:\"utmContent\";N;}", "template", null, null, 0, 0, "1", "a:1:{s:15:\"additional_data\";a:1:{s:4:\"list\";a:0:{}}}", null]:

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'message' cannot be null" at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 125 {"exception":"[object] (Doctrine\\DBAL\\Exception\\NotNullConstraintViolationException(code: 0): An exception occurred while executing 'INSERT INTO push_notifications (is_published, date_added, created_by, created_by_user, date_modified, modified_by, modified_by_user, checked_out, checked_out_by, checked_out_by_user, name, description, lang, url, heading, message, button, utm_tags, notification_type, publish_up, publish_down, read_count, sent_count, mobile, mobileSettings, category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [1, \"2023-02-03 08:15:14\", 1, \"Admin Mautic\", null, null, null, null, null, null, \"test\", null, \"en\", null, \"test\", null, null, \"a:4:{s:9:\\\"utmSource\\\";N;s:9:\\\"utmMedium\\\";N;s:11:\\\"utmCampaign\\\";N;s:10:\\\"utmContent\\\";N;}\", \"template\", null, null, 0, 0, \"1\", \"a:1:{s:15:\\\"additional_data\\\";a:1:{s:4:\\\"list\\\";a:0:{}}}\", null]:

SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'message' cannot be null at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php:125, Doctrine\\DBAL\\Driver\\PDO\\Exception(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'message' cannot be null at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDO/Exception.php:18, PDOException(code: 23000): SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'message' cannot be null at /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php:117)
[stacktrace]
#0 /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(182): Doctrine\\DBAL\\Driver\\AbstractMySQLDriver->convertException('An exception oc...', Object(Doctrine\\DBAL\\Driver\\PDO\\Exception))
#1 /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/DBALException.php(159): Doctrine\\DBAL\\DBALException::wrapException(Object(Doctrine\\DBAL\\Driver\\PDO\\MySQL\\Driver), Object(Doctrine\\DBAL\\Driver\\PDO\\Exception), 'An exception oc...')
#2 /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php(2226): Doctrine\\DBAL\\DBALException::driverExceptionDuringQuery(Object(Doctrine\\DBAL\\Driver\\PDO\\MySQL\\Driver), Object(Doctrine\\DBAL\\Driver\\PDO\\Exception), 'INSERT INTO pus...', Array)
#3 /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Statement.php(185): Doctrine\\DBAL\\Connection->handleExceptionDuringQuery(Object(Doctrine\\DBAL\\Driver\\PDO\\Exception), 'INSERT INTO pus...', Array, Array)
#4 /var/www/html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Statement.php(226): Doctrine\\DBAL\\Statement->execute(NULL)
#5 /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php(278): Doctrine\\DBAL\\Statement->executeStatement()
#6 /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(1133): Doctrine\\ORM\\Persisters\\Entity\\BasicEntityPersister->executeInserts()
#7 /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php(430): Doctrine\\ORM\\UnitOfWork->executeInserts(Object(Doctrine\\ORM\\Mapping\\ClassMetadata))
#8 /var/www/html/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php(403): Doctrine\\ORM\\UnitOfWork->commit(Object(Mautic\\NotificationBundle\\Entity\\Notification))
#9 /var/www/html/app/bundles/CoreBundle/Entity/CommonRepository.php(803): Doctrine\\ORM\\EntityManager->flush(Object(Mautic\\NotificationBundle\\Entity\\Notification))
#10 /var/www/html/app/bundles/CoreBundle/Model/FormModel.php(108): Mautic\\CoreBundle\\Entity\\CommonRepository->saveEntity(Object(Mautic\\NotificationBundle\\Entity\\Notification))
#11 /var/www/html/app/bundles/NotificationBundle/Controller/MobileNotificationController.php(291): Mautic\\CoreBundle\\Model\\FormModel->saveEntity(Object(Mautic\\NotificationBundle\\Entity\\Notification))
#12 /var/www/html/app/bundles/CoreBundle/Controller/CommonController.php(458): Mautic\\NotificationBundle\\Controller\\MobileNotificationController->newAction(Object(Mautic\\NotificationBundle\\Entity\\Notification), '')
#13 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(159): Mautic\\CoreBundle\\Controller\\CommonController->executeAction('new', 0, 0, '')
#14 /var/www/html/vendor/symfony/http-kernel/HttpKernel.php(81): Symfony\\Component\\HttpKernel\\HttpKernel->handleRaw(Object(Symfony\\Component\\HttpFoundation\\Request), 1)
#15 /var/www/html/vendor/symfony/http-kernel/Kernel.php(201): Symfony\\Component\\HttpKernel\\HttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#16 /var/www/html/app/AppKernel.php(105): Symfony\\Component\\HttpKernel\\Kernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#17 /var/www/html/app/middlewares/CORSMiddleware.php(82): AppKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#18 /var/www/html/app/middlewares/HSTSMiddleware.php(35): Mautic\\Middleware\\CORSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#19 /var/www/html/app/middlewares/CatchExceptionMiddleware.php(34): Mautic\\Middleware\\HSTSMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#20 /var/www/html/app/middlewares/Dev/IpRestrictMiddleware.php(52): Mautic\\Middleware\\CatchExceptionMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#21 /var/www/html/app/middlewares/VersionCheckMiddleware.php(58): Mautic\\Middleware\\Dev\\IpRestrictMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#22 /var/www/html/app/middlewares/TrustMiddleware.php(42): Mautic\\Middleware\\VersionCheckMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#23 /var/www/html/vendor/stack/builder/src/Stack/StackedHttpKernel.php(23): Mautic\\Middleware\\TrustMiddleware->handle(Object(Symfony\\Component\\HttpFoundation\\Request), 1, true)
#24 /var/www/html/vendor/stack/run/src/Stack/run.php(13): Stack\\StackedHttpKernel->handle(Object(Symfony\\Component\\HttpFoundation\\Request))
#25 /var/www/html/index_dev.php(25): Stack\
un(Object(Stack\\StackedHttpKernel))
#26 {main}
"} {"hostname":"mautic-web","pid":2411}```

**Comment by mabumusa1** [comment]
One minor thing, the title of the page needs adjusting 

image

**Comment by mabumusa1** [comment]
There is an issue to look for with the validation https://github.com/mautic/mautic/issues/11922

**Comment by TechBill** [comment]
Tested PR - All passed for me. 

```
/s/mobile_notifications List
Does page load and render correctly with and without any notifications present
Sorting columns work correctly
Filtering work correctly
```
**PASSED**


```
/s/mobile_notifications/new New
Can new notifications be created
```
**PASSED**

```
/s/mobile_notifications/view/{id} Details
Page load/render correctly with no errors
```
**PASSED**


```
/s/mobile_notifications/edit/{id} Details
Can you edit the notification
```
**PASSED**

**Comment by mabumusa1** [comment]
@allcontributors please add @TechBill  for userTesting

**Review by mabumusa1 — DISMISSED** [review]
I tested this PR & @TechBill did as well. I think this PR is good to go

**Comment by mabumusa1** [comment]
merging this PR when CI turns green

**Comment by mabumusa1** [comment]
@JoshuaEstes minor issue with the file names in the test to be fixed, then this is ready to be merged

**Review by mabumusa1 — APPROVED** [review]
The typo got fixed, this is ready to be merged

**Comment by mabumusa1** [comment]
@escopecz @RCheesley @JoshuaEstes On the final test before merging this PR, one of the tests [failed](https://github.com/mautic/mautic/actions/runs/4137324896/jobs/7153084201), this [package](https://github.com/php-http/discovery/releases/tag/1.15.0) released a new version. Because we do not have `composer.lock` in this testing step, composer is getting the latest version which should be added to `allow-plugins` section in `composer.json`

We are allowing this plugin in this PR

**Pull request merged**